File: Princy_Ballabh/Day_10.java
public class Day_10{

    public static void main(String args[]){
        int arr[]={4,5,7};
        System.out.println("Number of possible binary trees: "+trees(arr));
    }

    public static int trees(int[] arr) {
        return bintrees(arr, 0, arr.length - 1);
    }

    public static int bintrees(int[] arr,int s,int e){
        if(s>e){
            return 1;
        }
        int count=0;
        for(int i=s;i<=e;i++){
            int lc=bintrees(arr,s,i-1);
            int rc=bintrees(arr,i+1,e);
            count+=lc*rc;
        }
        return count;
    }   
}

File: Mohammad-Abbas/Day10.cpp
#include <iostream>
using namespace std;
int factorial(int n){int factorial=1;
    for(int i=1;i<=n;i++){
        factorial*=i;
    }
    return factorial;
}
int main(){
int arr[]={4,5,7};
int n=3;
    //Using Catalan Number 2n!/n!*(n+1)!
int ans=factorial(2*n)/(factorial(n)*factorial(n+1));
cout<<"Number of Binary Trees are:"<<ans<<endl;
return 0;

}